source/include/ble/cs_BleCentral.h

```diff
@@ -0,0 +1,178 @@
+	 *
+	 * @return ERR_BUSY                An operation is in progress (discovery, read, write, connect, disconnect).
+	 * @return ERR_WAIT_FOR_SUCCESS    When the read is started. Wait for EVT_BLE_CENTRAL_READ_RESULT.
+	 */
+	cs_ret_code_t read(uint16_t handle);
+
+	/**
+	 * Write data to a characteristic.
+	 *
+	 * @param[in] handle               The characteristic handle to write to. The handle was received during discovery.
+	 * @param[in] data                 Pointer to data, which will be copied.
+	 * @param[in] len                  Length of the data to write.
+	 *
+	 * @return ERR_BUFFER_TOO_SMALL    The data size is too large.
+	 * @return ERR_BUSY                An operation is in progress (discovery, read, write, connect, disconnect).
+	 * @return ERR_WAIT_FOR_SUCCESS    When the write is started. Wait for EVT_BLE_CENTRAL_WRTIE_RESULT.
+	 */
+	cs_ret_code_t write(uint16_t handle, const uint8_t* data, uint16_t len);
+
+	/**
+	 * Check whether this crownstone is connected as central.
+	 */
+	bool isConnected();
+
+	/**
+	 * Check whether an operation is in progress.
+	 */
+	bool isBusy();
+
+private:
+	enum class Operation: uint8_t {
+		NONE,
+		CONNECT,
+		DISCONNECT,
+		DISCOVERY,
+		READ,
+		WRITE
+	};
+
+	BleCentral();
+	BleCentral(BleCentral const&);
+	void operator=(BleCentral const &);
+
+	/**
+	 * Buffer used for reading and writing.
+	 */
+	uint8_t _buf[200];
+
+	/**
+	 * How much data is actually in the buffer.
+	 */
+	uint16_t _bufDataSize = 0;
+
+	uint16_t _connectionHandle = BLE_CONN_HANDLE_INVALID;
+
+	ble_db_discovery_t _discoveryModule;
+
+	/**
+	 * Current MTU.
+	 */
+	uint16_t _mtu = BLE_GATT_ATT_MTU_DEFAULT;
+
+	/**
+	 * Operation in progress.
+	 */
+	Operation _currentOperation = Operation::NONE;
+
+	/**
+	 * Writes the next chunk of a long write.
+	 */
+	cs_ret_code_t nextWrite(uint16_t handle, uint16_t offset);
+
+	/**
+	 * Finalize an operation.
+	 */
+	void finalizeOperation(Operation operation, cs_ret_code_t retCode);
+	void finalizeOperation(Operation operation, uint8_t* data, uint8_t dataSize);
+	void sendOperationResult(event_t& event);
+
+	/**
+	 * Event handlers.
+	 */
+	void onGapEvent(uint16_t evtId, const ble_gap_evt_t& event);
+	void onGattCentralEvent(uint16_t evtId, const ble_gattc_evt_t& event);
+
+	void onConnect(uint16_t connectionHandle, const ble_gap_evt_connected_t& event);
+	void onDisconnect(const ble_gap_evt_disconnected_t& event);
+	void onGapTimeout(const ble_gap_evt_timeout_t& event);
+
+	void onMtu(uint16_t gattStatus, const ble_gattc_evt_exchange_mtu_rsp_t& event);
+	void onRead(uint16_t gattStatus, const ble_gattc_evt_read_rsp_t& event);
+	void onWrite(uint16_t gattStatus, const ble_gattc_evt_write_rsp_t& event);
+
+
+public:
+	/**
+	 * Internal usage: to be called on BLE events.
+	 */
+	void onBleEvent(const ble_evt_t* event);
+
+	/**
+	 * Internal usage.
+	 */
+	void onDiscoveryEvent(ble_db_discovery_evt_t* event);
```

Comment:
as all event listeners use `event&` as parameter type, let's change these parameter types to `ble_evt_t&` and `ble_db_discovery_evt_t&` for uniformity.